### PR TITLE
[Torch Dialect] add folder for aten.Int.float

### PR DESCRIPTION
**Pull request by qingyunqu**

### CODE REVIEW

File: test/Dialect/Torch/canonicalize.mlir [hunk 1]
```diff
@@ -1253,6 +1253,15 @@ func.func @torch.aten.Int.Tensor(%arg0: !torch.int) -> !torch.int {
   return %scalar : !torch.int
 }
 
+// CHECK-LABEL:   func.func @torch.aten.Int.float() -> !torch.int {
+// CHECK:             %[[NUM:.*]] = torch.constant.int 1
+// CHECK:             return %[[NUM]] : !torch.int
+func.func @torch.aten.Int.float() -> !torch.int {
```
Comment (ramiro050) [inline test/Dialect/Torch/canonicalize.mlir:1259]: Can you also add an e2e test?

**Review by ramiro050 — CHANGES_REQUESTED** [review]
Thanks!

File: test/Dialect/Torch/canonicalize.mlir (hunk 1 above)
Comment (ZihengJiang) [reply test/Dialect/Torch/canonicalize.mlir:1259]: Hi Ramiro, it seems hard to add a MHLO conversion for a scalar integer to enable a e2e test. Also, the previous folder PRs all only do canonicalize test: https://github.com/llvm/torch-mlir/pull/1691, https://github.com/llvm/torch-mlir/pull/1257

Comment (qingyunqu) [reply test/Dialect/Torch/canonicalize.mlir:1259]: I add the e2e test. The torch.constant.int‘s value attribute is Singed Integer, while the  arith.constant only accept Signless Integer. So I add the type conversion when lowering torch.constant.int to arith.constant.

Comment (qingyunqu) [reply test/Dialect/Torch/canonicalize.mlir:1259]: The CI failed in strange way. Could you rerun it? Thanks.

**Review by ramiro050 — APPROVED** [review]
Thanks for adding the right conversion to arith!

**Pull request merged**